server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/management/DelayedSingleRetryRequestExecutor.java

```diff
@@ -0,0 +1,60 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.management;
+
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DelayedSingleRetryRequestExecutor implements RequestExecutor {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DelayedSingleRetryRequestExecutor.class);
+    public static final int TEN_MS_RETRY = 10;
+
+    @Override
+    public void execute(Runnable runnable) {
+        try {
```

Comment:
Here, maybe something like that:
```
        await().pollInterval(TEN_MS_RETRY)
                .atMost(TEN_MS_RETRY)
                .until(() -> {
                        runnable.run();
                        return true;
                    });
```

Reply:
It hides the idea of doing one retry. (you have to trust awaitability for it). Not to mention the ten MS delay you add upfront each call.

BTW how would you retrieve the value of the supplier 2 lines below?

If we have to choose a lib for it, then `com.nurkiewicz.asyncretry;`.

But I must admit that for such a simple strategy, coding *by hand* does not seem a bad option either. IMO let's choose the easy option and enhance when we need something more robust.

Reply:
writing it by end would make sense if we didn't solved the problem already with RetryExecutorUtil.retryOnExceptions
but in this case you duplicate something we already have

Reply:
moreover, you didn't even bother to write tests on this code !

Reply:
Done